lib/nurse/dependency_container.rb

```diff
@@ -0,0 +1,46 @@
+module Nurse
+  class DependencyContainer
+    class UndefinedDependency < RuntimeError; end
+
+    def define(dependency, &block)
+      definitions[to_key(dependency)] = block
+      self
+    end
+
+    def defined?(dependency)
+      definitions.has_key?(to_key(dependency))
+    end
+
+    def get(dependency)
+      return nil unless self.defined?(dependency)
+
+      key = to_key(dependency)
+
+      unless instances.has_key?(key)
+        instances[key] = definitions[key].call
+      end
+
+      instances[key]
+    end
+
+    def fetch(dependency, &block)
```

Comment:
Unused method argument - `block`. If it's necessary, use `_` or `_block` as an argument name to indicate that it won't be used.